libs/ssl-config/src/test/java/org/elasticsearch/common/ssl/SslDiagnosticsTests.java

```diff
@@ -63,7 +63,8 @@ public void testDiagnosticMessageWhenServerProvidesAFullCertChainThatIsTrusted()
         final String message = SslDiagnostics.getTrustDiagnosticFailure(chain, SslDiagnostics.PeerType.SERVER, session,
             "xpack.http.ssl", trustIssuers);
         assertThat(message, Matchers.equalTo("failed to establish trust with server at [192.168.1.1];" +
-            " the server provided a certificate with subject name [CN=cert1] and fingerprint [3bebe388a66362784afd6c51a9000961a4e10050];" +
+            " the server provided a certificate with subject name [CN=cert1] and fingerprint [3bebe388a66362784afd6c51a9000961a4e10050] and no keyUsage and no extendedKeyUsage;" +
+            " the session supports the cipher suite [<unknown cipherSuite>] and the protocol [<unknown protocol>];" +
```

Comment:
We need some of the tests to have real values for all of these.
As it is we're only testing the "no value" cases, which is the least important case. 

That will require regenerating some of the certs (and updating the fingerprints) to have key usage properties.
(see libs/ssl-config/src/test/resources/certs/README.txt)
and/or changing `mockCertificateWithIssuer` and `session` to provide mock values.

Reply:
I've edited the `mockCertificateWithIssuer` and `session` methods to provide mock values. I attempted to regenerate the cert yesterday without much success, so went with the mocking. Let me know if this works.